### PR TITLE
Solr 8.11.2 update

### PR DESCRIPTION
**Pull request by DonRichards**
This is untested.

I think this would work but I'm still testing it.

### CODE REVIEW

File: solr/README.md [hunk 1]
```diff
@@ -42,3 +42,30 @@ additional settings, volumes, ports, etc.
...
 [Solr Logging]: https://lucene.apache.org/solr/guide/7_1/configuring-logging.html
 [Solr]: https://lucene.apache.org/solr/
+
+## Changing versions
+There is 3 values you need to update/change the version. 
+1. Solr version: found at [archive.apache.org](https://archive.apache.org/dist/lucene/solr)
+1. SOLR_KEYS: Generated using GPG and the acs file
+1. SOLR_FILE_SHA256: sha256sum of the tgz file
+
+```dockerfile
```
Comment (nigelgbanks) [inline solr/README.md:52]: Same as above.

File: solr/Dockerfile [hunk 2]
```diff
@@ -3,11 +3,13 @@ ARG repository=local
 ARG tag=latest
 FROM --platform=$BUILDPLATFORM ${repository}/download:${tag} AS download
 
+ENV SOLR_VERSION="8.11.2"
+ENV SOLR_KEYS="86EDB9C33B8517228E88A8F93E48C0C6EF362B9E"
```
Comment (nigelgbanks) [inline solr/Dockerfile:7]: `SOLR_KEYS` is not used anywhere here?

File: solr/README.md [hunk 3]
```diff
@@ -42,3 +42,30 @@ additional settings, volumes, ports, etc.
 [Solr Documentation]: https://lucene.apache.org/solr/guide/7_1/
 [Solr Logging]: https://lucene.apache.org/solr/guide/7_1/configuring-logging.html
 [Solr]: https://lucene.apache.org/solr/
+
+## Changing versions
+There is 3 values you need to update/change the version. 
+1. Solr version: found at [archive.apache.org](https://archive.apache.org/dist/lucene/solr)
+1. SOLR_KEYS: Generated using GPG and the acs file
```
Comment (nigelgbanks) [inline solr/README.md:49]: Same as above.

File: solr/README.md [hunk 4]
```diff
@@ -42,3 +42,30 @@ additional settings, volumes, ports, etc.
 [Solr Documentation]: https://lucene.apache.org/solr/guide/7_1/
 [Solr Logging]: https://lucene.apache.org/solr/guide/7_1/configuring-logging.html
 [Solr]: https://lucene.apache.org/solr/
+
+## Changing versions
```
Comment (nigelgbanks) [inline solr/README.md:46]: Please add a space after the header for markdown standards.

**Review by nigelgbanks — CHANGES_REQUESTED** [review]
I have a few questions that might affect the upcoming release, and would require us to include some more info in the release notes, or do a minor/major bump.

1. Are any changes required on the Drupal side after this version bump?
2. Will this have an effect on someone's existing index (e.g. their `solr-data` Docker volume) if they were to switch images.
3.  Would they have to rebuild the index if it does?

Thanks!

File: solr/Dockerfile (hunk 2 above)
Comment (DonRichards) [reply solr/Dockerfile:7]: No, this is an attempt to make it easier/consistent for developers to update without needing to modify strings embedded within logic. By consistent I'm assuming this is the right way to approach this and modify other Dockerfiles in the future to match.

File: solr/README.md (hunk 4 above)
Comment (DonRichards) [reply solr/README.md:46]: A trailing space character after "## Changing versions^"?

Comment (nigelgbanks) [reply solr/README.md:46]: Sorry, newline after the header.

File: solr/Dockerfile [hunk 5]
```diff
@@ -3,11 +3,13 @@ ARG repository=local
 ARG tag=latest
 FROM --platform=$BUILDPLATFORM ${repository}/download:${tag} AS download
 
+ENV SOLR_VERSION="8.11.2"
```
[the remark this answers is not in the document]
Comment (DonRichards) [reply solr/Dockerfile:6]: Any suggestions for a replacement? Should I switch it to **ARG**?

**Comment by DonRichards** [comment]
@nigelgbanks To answer your original questions
1. Are any changes required on the Drupal side after this version bump?
> Not sure. It doesn't work for me yet so I imagine it might.

2. Will this have an effect on someone's existing index (e.g. their solr-data Docker volume) if they were to switch images.
> I'm glad you asked. I just looked it up [here](https://solr.apache.org/guide/8_0/major-changes-in-solr-8.html#rolling-upgrades-with-solr-8) and yes there is. Not sure how to integrate this into the codebase 

3. Would they have to rebuild the index if it does?
> They suggest always reindexing after an upgrade, I would assume a rebuild would be needed.

With all of that said, I'm not sure what the next steps should be for this PR.

File: solr/Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (nigelgbanks) [reply solr/Dockerfile:6]: `ARG` seems fine, we can change the other images for consistency as part of another issue (I'll create one).

[the remark this answers is not in the document]
Comment (nigelgbanks) [reply solr/Dockerfile:6]: https://github.com/Islandora-Devops/isle-buildkit/issues/245

File: solr/Dockerfile (hunk 2 above)
Comment (nigelgbanks) [reply solr/Dockerfile]: If it's not being used, let's just remove it. As it's an extra step that has no effect.

File: solr/Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (DonRichards) [reply solr/Dockerfile]: OK, great. Thanks for adding the ticket. Changes made.

**Comment by nigelgbanks** [comment]
> With all of that said, I'm not sure what the next steps should be for this PR.

At the least, we'll have to include it in the release notes. For `isle-dc` I'm not certain, there is a good solution as people can / are encouraged to maintain the `.env` file outside of source control, and we shouldn't be triggering re-builds of indexes if it's not appropriate.

**Comment by DonRichards** [comment]
Sorry, I just realized I was clicking "Resolved" on your comments as though they were something for me to "resolve". lol

**Comment by nigelgbanks** [comment]
> Sorry, I just realized I was clicking "Resolved" on your comments as though they were something for me to "resolve". lol

Haha no worries, after you have pushed the fix you can mark them as resolved.

File: solr/Dockerfile (hunk 2 above)
Comment (nigelgbanks) [reply solr/Dockerfile]: The documentation in the README.md should also get updated to not reference `SOLR_KEYS` after it's removed.

**Comment by DonRichards** [comment]
OK, I think I made all of the suggested changes. But to recap, on the release we should comment on how there is documentation from Solr on how to upgrade the index when upgrading Solr from 7 to 8. We should give an example command that the community has tested to upgrade Solr's index like the example they gave but adjusted for isle-dc

`./bin/solr start -c -Dsolr.http1=true -z localhost:2481/solr -s /path/to/solr/home`

They also had one last comment that we should probably spell out how to do it manually
> When all nodes have been upgraded to 8.0, restart each one without the -Dsolr.http1 parameter.

File: solr/Dockerfile (hunk 2 above)
Comment (DonRichards) [reply solr/Dockerfile]: OK, done

**Comment by DonRichards** [comment]
@nigelgbanks Do you see what's making the tests fail? I don't see what I need to change to get it to pass.

**Comment by nigelgbanks** [comment]
Looks to be the same `ps` [issue](https://github.com/Islandora-Devops/isle-buildkit/issues/243):

```bash
BusyBox v1.34.1 (2021-11-23 00:57:35 UTC) multi-call binary.

Usage: ps [-o COL1,COL2=HEADER] [-T]

Show list of processes

        -o COL1,COL2=HEADER     Select columns for display
        -T                      Show threads
This script relies on a version of ps that supports the -p flag.

Please install a POSIX compliant version and try again.
```

https://issues.apache.org/jira/browse/SOLR-16191

**Comment by nigelgbanks** [comment]
Looks like I read the ticket wrong, on reviewing their actual changes made for 8.11.2 it seems like they now have a hard requirement on using GNU version of `ps` rather than supporting both.

**Review by nigelgbanks — APPROVED** [review]

**Comment by nigelgbanks** [comment]
@DonRichards I've changed this to a draft just to prevent it from going in before we cut a minor release (which should be pretty soon). Subsequently, it should be good to go into the next major release, for which we'll document the need for folks to rebuild their `solr` index.

**Comment by DonRichards** [comment]
@nigelgbanks are you good with this dropping the "draft" status now?

**Comment by nigelgbanks** [comment]
@DonRichards whoops yea we'll start working on the 2.0.0 release.

**Comment by nigelgbanks** [comment]
@DonRichards I've raised a ticket, so I don't forget about updating the release notes.

**Pull request merged**